test/unit/spec/util/jobLogViewerSpec.js

```diff
@@ -48,217 +222,493 @@ define(['util/jobLogViewer', 'common/runtime'], (JobLogViewer, Runtime) => {
+                    });
+
+                    await this.jobLogViewerInstance.start({
+                        node: this.node,
                         jobId: jobId,
-                    },
-                    key: {
-                        type: 'job-status',
-                    },
-                };
-
-            runtimeBus.on('request-job-status', (msg) => {
-                expect(msg).toEqual({ jobId: jobId });
-                runtimeBus.send(jobData, channelData);
+                    });
+
+                    return TestUtil.waitForElementChange(
+                        this.node.querySelector('[data-element="log-panel"]')
+                    ).then(() => {
+                        expect(
+                            this.node.querySelector('[data-element="log-panel"]').textContent
+                        ).toBe('Job is queued; logs will be available when the job is running.');
+                    });
+                });
             });
 
-            runtimeBus.on('request-job-update', (msg) => {
-                expect(msg).toEqual({ jobId: jobId });
-                runtimeBus.send(jobData, channelData);
+            // running job: job logs are updated as they are received
+            it('Should render job logs whilst job is running', async function () {
+                const jobState = jobsByStatus['running'];
+                const jobId = jobState.job_id;
 
-                setTimeout(() => {
-                    const panel = hostNode.querySelector('[data-element="log-panel"]');
-                    expect(panel.children.length).toEqual(1);
-                    expect(panel.children[0].innerHTML).toContain('Job is queued'); //, logs will be available when the job is running.');
-                    done();
-                }, 500);
+                // lines to return each time there is a request for the latest logs
+                // first request - 0 lines; second: 2 log lines; third: same as second; last: all log lines.
+                const logs = [[], logLines.slice(0, 2), logLines.slice(0, 2), logLines]; //
+                let acc = 0;
+
+                this.runtimeBus.on('request-job-status', (msg) => {
+                    expect(msg).toEqual({ jobId: jobId });
+                    this.runtimeBus.send(...formatStatusMessage(jobState));
+                });
+
+                this.runtimeBus.on('request-job-update', (msg) => {
+                    expect(msg).toEqual({ jobId: jobId });
+                    this.runtimeBus.send(...formatStatusMessage(jobState));
+                });
+
+                await this.jobLogViewerInstance.start({
+                    node: this.node,
+                    jobId: jobId,
+                });
+
+                return new Promise((resolve) => {
+                    this.runtimeBus.on('request-latest-job-log', (msg) => {
+                        expect(msg).toEqual({ jobId: jobId, options: {} });
+                        const logUpdate = logs[acc];
```

Comment:
Every time `request-latest-job-log` is called, the next entry in the `logs` array is sent to the widget.